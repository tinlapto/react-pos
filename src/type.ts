export type Item = {
  id: number;
  title: string;
  photo: string;
  price: number;
};

export type Topping = {
  id: number;
  name: string;
};

export type Size = {
  id: number;
  name: string;
};

export type ItemDetail = Item & {
  toppings: Topping[];
  sizes: Size[];
};

export type CartItem = {
  id: number;
  title: string;
  price: number;
  toppings: Topping[];
  sizes: Size[];
  selectedToppings: number[];
  selectedSize: number;
  count: number;
};
