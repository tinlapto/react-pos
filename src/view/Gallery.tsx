import Card from "../component/Card";
import { useGetItems } from "../api";

const Gallery = ({ onClick: onCardClick }: { onClick: Function }) => {
  const { data: items = [] } = useGetItems();
  return (
    <div style={{ display: "flex", flexWrap: "wrap" }}>
      {items.map(({ id, title, price, photo }) => (
        <Card
          key={id}
          title={`${title} - $${price}`}
          photo={photo}
          onClick={() => {
            onCardClick(id);
          }}
        />
      ))}
    </div>
  );
};

export default Gallery;
