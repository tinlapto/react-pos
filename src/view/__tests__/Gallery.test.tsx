import { render, screen, fireEvent } from "@testing-library/react";
import Gallery from "../Gallery";
import { Item } from "../../type";

const pizzas: Item[] = [
  { id: 1, title: "test name", price: 10, photo: "test photo" },
];

jest.mock("../../api", () => ({
  useGetItems: () => ({
    data: pizzas,
  }),
}));

describe("Gallery", () => {
  test("render", () => {
    const onClick = jest.fn();
    const { asFragment } = render(<Gallery onClick={onClick} />);
    expect(asFragment()).toMatchSnapshot();
    fireEvent.click(screen.getByText("Choose"));
    expect(onClick).toBeCalled();
  });
});
