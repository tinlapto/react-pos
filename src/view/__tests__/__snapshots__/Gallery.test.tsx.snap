// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery render 1`] = `
<DocumentFragment>
  <div
    style="display: flex; flex-wrap: wrap;"
  >
    <div
      class="sc-bdvvtL egAXI"
    >
      <div
        class="sc-gsDKAQ gKCqzO"
      />
      <div
        class="sc-dkPtRN lmuKJx"
      >
        test name - $10
      </div>
      <button
        class="ant-btn ant-btn-default"
        style="width: 100%;"
        type="button"
      >
        <span>
          Choose
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
